### DISCUSSION TITLE
How can I mark the current EditorComponent tab as "dirty"?

### DISCUSSION THREAD

**Discussion by mister-teddy** [opensumi/core#2070] [Q&A]
I'm following this [guide](https://opensumi.com/en/docs/develop/module-apis/editor#example) to create a custom editor component for my IDE. How can I mark the current editor as dirty when the user updates these form fields (like when the text content of the file has changed 👇)

<img width="787" alt="Screenshot 2022-12-07 at 17 20 52" src="https://user-images.githubusercontent.com/29925961/206154160-0792802b-71f2-47f8-9524-e7a32e4982f9.png">

Thank you for your help 🙏

**Comment by mister-teddy** [comment]
And how to clear the dirty state afterwards? I think I'll put it here, is it alright? 😅 
```ts
registry.registerEditorComponentResolver('file', (resource, results) => {
  results.push({
    saveResource: async (resource, reason) => {
      // clear dirty state here
```

**Comment by erha19** [comment]
Hi @nguyenhongphat0,

As I know, you can use the `BrowserEditorContribution` to define the resource and how to provide its name, icon, whether it is edited ("dirty") or not when it is displayed on the editor tab, and the corresponding callback when the tab is closed, etc.

Some code samples on `opensumi/core` :

https://github.com/opensumi/core/blob/7144fa47a338293ceef845cc2080bc4a6860ff91/packages/keymaps/src/browser/keymaps.contribution.ts#L152-L154

You can register a resource provider to handle the tab 'dirty' state by defining a `getResourceDecoration` method, like below:

```ts
  getResourceDecoration() {
    return {
      dirty: true,
    };
  }
```

hope this can help you.

**Reply by erha19** [reply]
Update, I tried it and found that the  `getResourceDecoration` method has not been implemented in the `BrowserEditorContribution `. Currently, the EditorComponent type cannot support this feature.

**Reply by mister-teddy** [reply]
Thank you for your reply! I'm looking forward for the `getResourceDecoration` to be implemented. Please let me know then, thank you very much 🙏
